src/binaryen-c.h

```diff
@@ -1180,6 +1180,15 @@ BINARYEN_API void BinaryenSetFunctionTable(BinaryenModuleRef module,
                                            const char** funcNames,
                                            BinaryenIndex numFuncNames,
                                            BinaryenExpressionRef offset);
+BINARYEN_API int BinaryenIsFunctionTableImported(BinaryenModuleRef module);
+BINARYEN_API BinaryenIndex
+BinaryenGetFunctionTableNumSegments(BinaryenModuleRef module);
+BINARYEN_API BinaryenExpressionRef BinaryenGetFunctionTableSegmentOffset(
+  BinaryenModuleRef module, BinaryenIndex id);
+BINARYEN_API BinaryenIndex BinaryenGetFunctionTableSegmentDataLength(
+  BinaryenModuleRef module, BinaryenIndex id);
+BINARYEN_API const char**
+BinaryenGetFunctionTableSegmentData(BinaryenModuleRef module, BinaryenIndex id);
```

Comment:
The Memory API copies the data out instead of just returning a reference to the owned data. I think it would be safer to do that here, too, to prevent the user holding on to a dangling pointer if more data is added later and the underlying vector is reallocated.